pkg/volume/glusterfs/glusterfs.go

```diff
@@ -288,23 +288,51 @@ func (c *glusterfsUnmounter) TearDownAt(dir string) error {
 
 func (b *glusterfsMounter) setUpAtInternal(dir string) error {
 	var errs error
-
 	options := []string{}
+	hasLogFile := false
+	log := ""
+
 	if b.readOnly {
 		options = append(options, "ro")
+
 	}
 
-	p := path.Join(b.glusterfs.plugin.host.GetPluginDir(glusterfsPluginName), b.glusterfs.volName)
-	if err := os.MkdirAll(p, 0750); err != nil {
-		return fmt.Errorf("failed to create directory %v: %v", p, err)
+	// Check logfile has been provided by user, if provided, use that as the log file.
+	for i, userOpt := range b.mountOptions {
+		if dstrings.Contains(userOpt, "log-file") {
```

Comment:
`HasPrefix()` would be probably more safe here

Reply:
Sure. will address it in next patch set.

Reply:
Done.